frontend/packages/ux-editor/src/components/config/editModal/EditOptions/OptionTabs/EditTab/EditTab.tsx

```diff
@@ -1,102 +1,104 @@
-import React, { useState } from 'react';
+import React, { createRef } from 'react';
+import {
+  StudioAlert,
+  StudioButton,
+  StudioErrorMessage,
+  StudioSpinner,
+  usePrevious,
+} from '@studio/components';
 import { useTranslation } from 'react-i18next';
-import { StudioDeleteButton, StudioErrorMessage } from '@studio/components';
-import { AddManualOptionsModal } from './AddManualOptionsModal';
-import { OptionListSelector } from './OptionListSelector';
-import { OptionListUploader } from './OptionListUploader';
-import { OptionListEditor } from './/OptionListEditor';
+import { useUpdate } from 'app-shared/hooks/useUpdate';
 import { useComponentErrorMessage } from '../../../../../../hooks';
+import { handleOptionsChange, updateComponentOptions } from './utils/utils';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useOptionListIdsQuery } from '../../../../../../hooks/queries/useOptionListIdsQuery';
 import type { IGenericEditComponent } from '../../../../componentConfig';
 import type { SelectionComponentType } from '../../../../../../types/FormComponent';
+import { OptionListSelector } from './OptionListSelector';
+import { OptionListUploader } from './OptionListUploader';
+import { OptionListEditor } from './OptionListEditor';
 import classes from './EditTab.module.css';
 
-type EditOptionChoiceProps = Pick<
+type EditTabProps = Pick<
   IGenericEditComponent<SelectionComponentType>,
   'component' | 'handleComponentChange'
 >;
 
-export function EditTab({
-  component,
-  handleComponentChange,
-}: EditOptionChoiceProps): React.ReactElement {
-  const initialComponentHasOptionList: boolean = !!component.optionsId || !!component.options;
-  const [componentHasOptionList, setComponentHasOptionList] = useState<boolean>(
-    initialComponentHasOptionList,
+export function EditTab({ component, handleComponentChange }: EditTabProps): React.ReactElement {
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { data: optionListIds, status } = useOptionListIdsQuery(org, app);
+  const previousComponent = usePrevious(component);
+  const dialogRef = createRef<HTMLDialogElement>();
+
+  useUpdate(() => {
+    if (!previousComponent.options && !!component.options) dialogRef.current.showModal();
+  }, [component, previousComponent]);
+
+  const optionsIdIsFromLibrary = optionListIds?.some(
+    (optionId: string) => optionId === component.optionsId,
   );
-  const errorMessage = useComponentErrorMessage(component);
+  const isOptionsIdReferenceId = !!component.optionsId && !optionsIdIsFromLibrary;
+  const isOptionsModifiable =
+    (!!component.optionsId && optionsIdIsFromLibrary) || !!component.options;
```

Comment:
Maybe these values either can be defined after the query has completed or in functions outside the component? 
Either way we would not need the `?` for the optionListIds (backend responds with empty array if app has no option lists).
And if defining as functions we can use them directly in the success return.